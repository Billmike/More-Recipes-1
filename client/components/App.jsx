import React from 'react';
import PropTypes from 'prop-types';
import { Route, Switch, Redirect } from 'react-router-dom';

import HomePage from './homepage';
import CatalogPage from './catalogPage';
import SingleRecipePage from './singleRecipePage';
import UserDashboardPage from './userRecipes';
import NotFoundPage from './notFoundPage';
import { decodeToken } from '../utils/authenticate';
import '../public/style.scss';

const propTypes = {
  location: PropTypes.shape(),
};

const defaultProps = {
  location: undefined,
};

/**
 * App component
 *
 * @returns {any} - Navigate to the matched route
 */
const App = () => (
  <Switch>
    <Route exact path="/" component={HomePage} />
    <Route exact path="/catalog" component={CatalogPage} />
    <Route exact path="/recipes/:recipeId" component={SingleRecipePage} />
    <Route
      path="/users/:userId/dashboard"
      render={props => (decodeToken()
        ? <UserDashboardPage {...props} />
        : <Redirect to={{
          pathname: '/',
          state: {
            from: props.location
          }
        }}
        />)
      }
    />
    <Route component={NotFoundPage} />
  </Switch>
);

App.propTypes = propTypes;
App.defaultProps = defaultProps;

export default App;
