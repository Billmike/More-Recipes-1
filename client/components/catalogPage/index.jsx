import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import actionCreators from '../../actions';
import Header from '../common/Header';
import Main from './Main';
import Footer from '../common/Footer';

const propTypes = {
  searchRecipe: PropTypes.func.isRequired,
  location: PropTypes.shape({
    state: PropTypes.shape({
      search: PropTypes.string,
      searchTerm: PropTypes.string,
    }),
  }),
  recipes: PropTypes.shape({
    searchResult: PropTypes.arrayOf(PropTypes.shape()),
  }).isRequired,
};

const defaultProps = {
  location: undefined,
};

/**
 * Class representing catalog page
 *
 * @class CatalogPage
 * @extends {Component}
 */
class CatalogPage extends Component {
  /**
   * Creates an instance of CatalogPage.
   *
   * @memberof CatalogPage
   */
  constructor() {
    super();
    this.state = {
      searchBy: 'name',
      searchTerm: '',
      searchedTerm: '',
      showText: false,
    };
  }

  /**
   * Component did mount lifecycle method
   *
   * @returns {any} Any
   * @memberof CatalogPage
   */
  componentDidMount() {
    const { location: { state }, searchRecipe } = this.props;
    window.scrollTo(0, 0);
    if (state) {
      searchRecipe(state.searchBy, state.searchTerm);
    }
  }

  /**
   * Component will receive props lifecycle method
   *
   * @returns {any} Sets state
   * @memberof CatalogPage
   */
  componentWillReceiveProps() {
    if (this.props.recipes.searchResult) {
      this.setState({
        showText: true,
      });
    }
  }

  /**
   * Function to handle change in search term
   *
   * @param {object} event - The event object
   *
   * @returns {any} Changes state to the current search term
   * @memberof CatalogPage
   */
  handleSearchChange = (event) => {
    const { target: { value } } = event;

    if (value === 'name') {
      this.setState(() => ({
        searchBy: value
      }));
    } else if (value === 'ingredients') {
      this.setState(() => ({
        searchBy: value,
      }));
    } else {
      this.setState(() => ({
        searchTerm: value
      }));
    }
  }

  /**
   * Function to handle submitting search term
   *
   * @param {object} event - The event object
   *
   * @returns {any} Submits the search term
   * @memberof CatalogPage
   */
  handleSubmitSearch = (event) => {
    event.preventDefault();
    const { searchBy, searchTerm } = this.state;
    const { searchRecipe } = this.props;

    this.setState(() => ({
      searchedTerm: searchTerm,
    }));

    searchRecipe(searchBy, searchTerm);
  }

  /**
   * Render method
   *
   * @returns {object} React element
   * @memberof CatalogPage
   */
  render() {
    return (
      <div>
        <header>
          <Header {...this.props} />
        </header>
        <main>
          <Main
            handleSearchChange={this.handleSearchChange}
            handleSubmitSearch={this.handleSubmitSearch}
            {...this.state}
            {...this.props}
          />
        </main>
        <footer>
          <Footer />
        </footer>

      </div>
    );
  }
}

/**
 * Function to map values from state to props
 *
 * @param {any} state - The state values
 *
 * @returns {object} - The mapped props
 */
const mapStateToProps = state => ({
  recipes: state.recipes,
  user: state.user,
});

/**
 * Function to map dispatch to props
 * Action creators are binded to the dispatch function
 *
 * @param {any} dispatch - The store dispatch function
 *
 * @returns {any} The mapped props
 */
const mapDispatchToProps = dispatch => (
  bindActionCreators(actionCreators, dispatch)
);

CatalogPage.propTypes = propTypes;
CatalogPage.defaultProps = defaultProps;

export default connect(mapStateToProps, mapDispatchToProps)(CatalogPage);
