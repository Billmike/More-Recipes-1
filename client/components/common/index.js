import Header from './Header';
import Footer from './Footer';
import RecipeImage from './RecipeImage';

export default {
  Header,
  Footer,
  RecipeImage,
};
