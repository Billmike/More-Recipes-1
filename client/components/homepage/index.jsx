import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import actionCreators from '../../actions';
import Header from '../common/Header';
import Footer from '../common/Footer';
import Main from './Main';

const propTypes = {
  recipes: PropTypes.shape({
    recipes: PropTypes.arrayOf(PropTypes.shape()),
  }).isRequired,
  retrieveRecipes: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func,
  }).isRequired,
};

/**
 * Class representing the home component
 *
 * @class Home
 *
 * @extends {Component}
 */
class Home extends Component {
  /**
   * Creates an instance of Home.
   *
   * @memberof Home
   */
  constructor() {
    super();
    this.state = {
      search: 'title',
      list: '',
    };
  }

  /**
   * Component did mount function
   *
   * @returns {object} The response from the server
   *
   * @memberof Home
   */
  componentDidMount() {
    const { recipes: { recipes } } = this.props;

    if (!recipes.length > 0) {
      this.props.retrieveRecipes();
    }
  }

  /**
   * Function to handle change in search term
   *
   * @param {object} event - The event object
   *
   * @returns {any} Changes state to the current search term
   *
   * @memberof Home
   */
  handleSearchChange = (event) => {
    const { target: { value } } = event;

    if (value === 'title') {
      this.setState(() => ({
        search: value
      }));
    } else if (value === 'ingredients') {
      this.setState(() => ({
        search: value,
      }));
    } else {
      this.setState(() => ({
        list: value
      }));
    }
  }

  /**
   * Function to handle submitting search term
   *
   * @param {object} event - The event object
   *
   * @returns {any} Submits the search term
   *
   * @memberof Home
   */
  handleSubmitSearch = (event) => {
    event.preventDefault();
    const { search, list } = this.state;

    if (list !== '') {
      this.props.history.push({
        pathname: '/catalog',
        search: `?search=${search}&list=${list}`,
      });
    }
  }

  /**
   * The render function
   *
   * @returns {object} React element
   *
   * @memberof Home
   */
  render() {
    return (
      <div className="page-body">
        <header>
          <Header {...this.props} />
        </header>
        <main>
          <Main
            handleSearchChange={this.handleSearchChange}
            handleSubmitSearch={this.handleSubmitSearch}
            {...this.state}
            {...this.props}
          />
        </main>
        <footer>
          <Footer />
        </footer>
      </div>
    );
  }
}

/**
 * Function to map values from state to props
 *
 * @param {any} state - The state values
 *
 * @returns {object} - The mapped props
 */
const mapStateToProps = state => ({
  recipes: state.recipes,
  user: state.user,
  userRecipes: state.userRecipes,
});

/**
 * Function to map dispatch to props
 * Action creators are binded to the dispatch function
 *
 * @param {any} dispatch - The store dispatch function
 *
 * @returns {any} The mapped props
 */
const mapDispatchToProps = dispatch => (
  bindActionCreators(actionCreators, dispatch)
);

Home.propTypes = propTypes;

export default connect(mapStateToProps, mapDispatchToProps)(Home);
