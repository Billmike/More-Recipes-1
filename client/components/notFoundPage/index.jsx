import React from 'react';
import { Link } from 'react-router-dom';

/**
 * Page not found component
 *
 * @returns {object} React element
 */
const NotFoundPage = () => (
  <div className="row">
    <nav>
      <div className="nav-wrapper deep-orange darken-4">
        <Link
          to="/"
          id="logo"
          className="brand-logo hide-on-small-only"
        >
          More-Recipes
        </Link>
      </div>
    </nav>
    <div className="row center-align">
      <div className="not-found">
        <p className="status-number">404</p>
        <p className="page-data">PAGE NOT FOUND</p>
        <i className="material-icons large">error</i>
      </div>
      <Link to="/">Go home</Link>
    </div>
  </div>

);

export default NotFoundPage;
