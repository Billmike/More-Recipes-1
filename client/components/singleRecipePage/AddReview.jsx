import React from 'react';
import PropTypes from 'prop-types';

const AddReview = (props) => {
  console.log(props);
  const { reviewContent, handleChange, handleSubmitReview } = props;
  return (
    <div className="row">
      <div className="col s12 m12 l8">
        <form
          className="add-review col s12 center-align"
          onSubmit={handleSubmitReview}
        >
          <div className="row">
            <div type="text" className="input-field col s12">
              <textarea
                id="new-review"
                type="text"
                value={reviewContent}
                className="materialize-textarea"
                onChange={handleChange}
                required
              />
              <label htmlFor="new-review">Add your review</label>
            </div>
          </div>
          <div className="row">
            <button
              type="submit"
              className={`btn waves-effect waves-light indigo accent-2
              white-text`}
            >
              Post Review
            </button>
          </div>
        </form>
      </div>
    </div>
  );
};

AddReview.propTypes = {
  reviewContent: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleSubmitReview: PropTypes.func.isRequired,
};

export default AddReview;
