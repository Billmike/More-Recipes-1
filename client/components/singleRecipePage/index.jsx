import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import isEmpty from 'lodash/isEmpty';

import actionCreators from '../../actions';
import Header from '../common/Header';
import Footer from '../common/Footer';
import Main from './Main';
import { getToken } from '../../utils/authenticate';

const propTypes = {
  singleRecipe: PropTypes.shape({
    recipe: PropTypes.shape({
      id: PropTypes.number,
    })
  }).isRequired,
  user: PropTypes.shape({
    isAuthenticated: PropTypes.bool.isRequired,
  }).isRequired,
  fetchRecipe: PropTypes.func.isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      recipeId: PropTypes.string,
    })
  }).isRequired,
  postReview: PropTypes.func.isRequired,
  upvoteRecipe: PropTypes.func.isRequired,
  downvoteRecipe: PropTypes.func.isRequired,
  setFavorite: PropTypes.func.isRequired,
};

/**
 * Class representing recipe details page
 *
 * @class Recipe
 * @extends {Component}
 */
class Recipe extends Component {
  /**
   * Creates an instance of Recipe.
   *
   * @param {any} props - This is the props param
   * @memberof Recipe
   */
  constructor(props) {
    super(props);
    this.state = {
      openSignup: false,
      openSignin: false,
      reviewContent: '',
    };
    this.handleOpenSignup = this.handleOpenSignup.bind(this);
    this.handleOpenSignin = this.handleOpenSignin.bind(this);
    this.handleClose = this.handleClose.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleAddReview = this.handleAddReview.bind(this);
    this.handleUpvote = this.handleUpvote.bind(this);
    this.handleDownvote = this.handleDownvote.bind(this);
    this.handleFavorite = this.handleFavorite.bind(this);
  }

  /**
   * Component did mount
   *
   * @returns {object} React element
   * @memberof Recipe
   */
  componentDidMount() {
    // Initialize materialize css parallax class
    $('.parallax').parallax();
    this.props.fetchRecipe(this.props.match.params.recipeId);
  }

  /**
   * Opens the signup modal
   *
   * @returns {object} Set open state to true
   * @memberof Recipe
   */
  handleOpenSignup() {
    this.setState({ openSignup: true });
  }

  /**
   * Opens the signin modal
   *
   * @returns {object} Set open state to true
   * @memberof Recipe
   */
  handleOpenSignin() {
    this.setState({ openSignin: true });
  }

  /**
   * Closes the modal
   *
   * @param {object} errors - The error object
   * @returns {object} Set open state to false
   * @memberof Recipe
   */
  handleClose() {
    this.setState({
      openSignup: false,
      openSignin: false
    });
  }

  /**
   * Funtion to handle review input state
   *
   * @param {string} event - The input value
   * @returns {string} The updated state
   * @memberof Recipe
   */
  handleChange(event) {
    this.setState({
      reviewContent: event.target.value,
    });
  }

  /**
   * Function to handle submission of review
   *
   * @param {string} event - The user review
   * @returns {func} Dispatch function
   * @memberof Recipe
   */
  handleAddReview(event) {
    event.preventDefault();
    const { user: { isAuthenticated } } = this.props,
      review = {
        content: this.state.reviewContent,
      };
    if (isAuthenticated && getToken() !== null) {
      this.props.postReview(
        this.props.match.params.recipeId, review
      )
        .then(() => {
          const { singleRecipe: { error } } = this.props;
          if (isEmpty(error)) {
            this.setState({
              reviewContent: '',
            });
          }
        });
    } else {
      this.handleOpenSignin();
    }
  }

  /**
   * Function to handle upvoting a recipe
   *
   * @returns {any} Call to upvote recipe action
   * @memberof Recipe
   */
  handleUpvote() {
    const { singleRecipe, upvoteRecipe } = this.props;
    const { recipe: { id } } = singleRecipe;

    upvoteRecipe(id);
  }

  /**
   * Function to handle downvoting a recipe
   *
   * @returns {any} Call to upvote recipe action
   * @memberof Recipe
   */
  handleDownvote() {
    const { singleRecipe, downvoteRecipe } = this.props;
    const { recipe: { id } } = singleRecipe;

    downvoteRecipe(id);
  }

  /**
   * Function to handle downvoting a recipe
   *
   * @returns {any} Call to action for favoriting a recipe
   * @memberof Recipe
   */
  handleFavorite() {
    const { singleRecipe, setFavorite } = this.props;
    const { recipe: { id } } = singleRecipe;

    setFavorite(id);
  }

  /**
   * Render method
   *
   * @returns {object} React element
   * @memberof Recipe
   */
  render() {
    const { singleRecipe } = this.props;

    return (
      <div>
        {!isEmpty(singleRecipe) &&
          <div>
            <header>
              <Header
                openSignup={this.state.openSignup}
                openSignin={this.state.openSignin}
                handleOpenSignup={this.handleOpenSignup}
                handleOpenSignin={this.handleOpenSignin}
                handleClose={this.handleClose}
                {...this.props}
              />
            </header>
            <main>
              <Main
                singleRecipe={singleRecipe}
                reviewContent={this.state.reviewContent}
                handleChange={this.handleChange}
                handleAddReview={this.handleAddReview}
                handleUpvote={this.handleUpvote}
                handleDownvote={this.handleDownvote}
                handleFavorite={this.handleFavorite}
              />
            </main>
            <footer>
              <Footer />
            </footer>
          </div>
        }
      </div>
    );
  }
}

/**
 * Function to map alues from state to props
 *
 * @param {object} state - The state values
 * @returns {object} - The mapped props
 */
const mapStateToProps = state => ({
  singleRecipe: state.singleRecipe,
  user: state.user,
});

/**
 * Function to map dispatch to props
 * Action creators are binded to the dispatch function
 *
 * @param {any} dispatch
 * @returns {any} The mapped props
 */
const mapDispatchToProps = dispatch => (
  bindActionCreators(actionCreators, dispatch)
);

Recipe.propTypes = propTypes;

export default connect(mapStateToProps, mapDispatchToProps)(Recipe);
