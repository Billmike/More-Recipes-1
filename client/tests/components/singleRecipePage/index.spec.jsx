import React from 'react';

import { Recipe } from '../../../components/singleRecipePage';
import userMockData from '../../__mocks__/user';

describe('Recipe component', () => {
  let spy;

  beforeEach(() => {
    localStorage.setItem(
      'token',
      userMockData.signinSuccessResponse.data.user.token
    );
  });
  afterEach(() => {
    spy.mockClear();
  });
  it('should render component', () => {
    const props = {
      singleRecipe: {
        isFetching: false,
        recipe: {},
        reviews: [],
      },
      user: {
        isAuthenticated: false,
      },
      fetchRecipe: jest.fn(),
      history: {
        push: jest.fn(),
      },
      match: {
        params: {
          recipeId: 2,
        },
      },
      postReview: jest.fn(),
      upvoteRecipe: jest.fn(),
      downvoteRecipe: jest.fn(),
      setFavorite: jest.fn(),
      fetchReviews: jest.fn(),
      clearReviews: jest.fn(),
      deleteReview: jest.fn(),
    };
    spy = jest.spyOn(Recipe.prototype, 'componentDidMount');
    const wrapper = shallow(<Recipe {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
    expect(spy).toHaveBeenCalled();
    expect(props.fetchRecipe).toHaveBeenCalled();
    expect(props.fetchReviews).toHaveBeenCalled();
    wrapper.setProps({ singleRecipe: { isFetching: true } });
    expect(wrapper.find('Spinner').exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({
      singleRecipe: {
        isFetching: false,
        hasMoreReviews: true,
        recipe: {
          id: 2,
          title: 'Beans',
          category: 'Main',
          description: 'Tasty beans',
          preparationTime: 85,
          ingredients: 'Beans, onions, palm oil, salt',
          directions: 'Do this. Do that',
          upvotes: 4,
          downvotes: 0,
          favorites: 8,
          views: 25,
        },
        reviews: [{
          content: 'Nice recipe',
        }],
      },
    });
    expect(wrapper.instance().state.offset).toEqual(1);
    expect(wrapper.find('Spinner').exists()).toBe(false);
    expect(wrapper.find('Main').exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
  it('should simulate instance methods call', () => {
    const props = {
      singleRecipe: {
        isFetching: false,
        recipe: {
          id: 2,
        },
        reviews: [],
      },
      user: {
        isAuthenticated: true,
      },
      fetchRecipe: jest.fn(),
      history: {
        push: jest.fn(),
      },
      match: {
        params: {
          recipeId: 2,
        },
      },
      postReview: jest.fn(() => Promise.resolve()),
      upvoteRecipe: jest.fn(),
      downvoteRecipe: jest.fn(),
      setFavorite: jest.fn(),
      fetchReviews: jest.fn(),
      clearReviews: jest.fn(),
      deleteReview: jest.fn(),
    };
    const event = {
      preventDefault: jest.fn(),
      target: {
        value: 'Change in input field',
      },
    };
    const wrapper = shallow(<Recipe {...props} />);
    spy = jest.spyOn(wrapper.instance(), 'handleChange');

    expect(wrapper).toBeDefined();
    wrapper.instance().handleChange(event);
    expect(spy).toBeCalled();
    wrapper.instance().handleAddReview(event);
    wrapper.instance().handleDeleteReview(2, 1);
    wrapper.instance().handleUpvote();
    wrapper.instance().handleDownvote();
    wrapper.instance().handleFavorite();
    wrapper.instance().handleViewMoreReviews();
    wrapper.instance().handleSearchCategory(event);
    expect(props.deleteReview).toBeCalled();
    expect(wrapper).toMatchSnapshot();
  });
  it('should clear review when component is been unmounted', () => {
    const props = {
      singleRecipe: {
        isFetching: false,
        recipe: {
          id: 2,
        },
        reviews: [],
      },
      user: {
        isAuthenticated: true,
      },
      fetchRecipe: jest.fn(),
      history: {
        push: jest.fn(),
      },
      match: {
        params: {
          recipeId: 2,
        },
      },
      postReview: jest.fn(),
      upvoteRecipe: jest.fn(),
      downvoteRecipe: jest.fn(),
      setFavorite: jest.fn(),
      fetchReviews: jest.fn(),
      clearReviews: jest.fn(),
      deleteReview: jest.fn(),
    };
    spy = jest.spyOn(Recipe.prototype, 'componentWillUnmount');
    const wrapper = shallow(<Recipe {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
    expect(spy).toBeCalled();
    expect(props.clearReviews).toBeCalled();
    expect(wrapper).toMatchSnapshot();
  });
});
