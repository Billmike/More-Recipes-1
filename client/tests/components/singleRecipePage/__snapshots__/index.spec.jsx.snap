// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe component should clear review when component is been unmounted 1`] = `
<div
  className="row"
>
  <div
    className="page-body"
  >
    <header>
      <Header
        clearReviews={[MockFunction]}
        deleteReview={[MockFunction]}
        downvoteRecipe={[MockFunction]}
        fetchRecipe={
          [MockFunction] {
            "calls": Array [
              Array [
                2,
              ],
            ],
          }
        }
        fetchReviews={
          [MockFunction] {
            "calls": Array [
              Array [
                2,
              ],
            ],
          }
        }
        handleSearchCategory={[Function]}
        history={
          Object {
            "push": [MockFunction],
          }
        }
        match={
          Object {
            "params": Object {
              "recipeId": 2,
            },
          }
        }
        postReview={[MockFunction]}
        setFavorite={[MockFunction]}
        singleRecipe={
          Object {
            "isFetching": false,
            "recipe": Object {
              "id": 2,
            },
            "reviews": Array [],
          }
        }
        upvoteRecipe={[MockFunction]}
        user={
          Object {
            "isAuthenticated": true,
          }
        }
      />
    </header>
    <main>
      <div>
        <Main
          handleAddReview={[Function]}
          handleChange={[Function]}
          handleDeleteReview={[Function]}
          handleDownvote={[Function]}
          handleFavorite={[Function]}
          handleUpvote={[Function]}
          handleViewMoreReviews={[Function]}
          reviewContent=""
          singleRecipe={
            Object {
              "isFetching": false,
              "recipe": Object {
                "id": 2,
              },
              "reviews": Array [],
            }
          }
        />
      </div>
      <ToastContainer
        autoClose={5000}
        bodyClassName=""
        className={null}
        closeButton={
          <DefaultCloseButton
            ariaLabel="close"
          />
        }
        closeOnClick={true}
        hideProgressBar={false}
        newestOnTop={false}
        pauseOnHover={true}
        position="top-right"
        progressClassName=""
        style={null}
        toastClassName=""
        transition={[Function]}
      />
    </main>
    <footer>
      <Footer />
    </footer>
  </div>
</div>
`;

exports[`Recipe component should clear review when component is been unmounted 2`] = `
<div
  className="row"
>
  <div
    className="page-body"
  >
    <header>
      <Header
        clearReviews={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
          }
        }
        deleteReview={[MockFunction]}
        downvoteRecipe={[MockFunction]}
        fetchRecipe={
          [MockFunction] {
            "calls": Array [
              Array [
                2,
              ],
            ],
          }
        }
        fetchReviews={
          [MockFunction] {
            "calls": Array [
              Array [
                2,
              ],
            ],
          }
        }
        handleSearchCategory={[Function]}
        history={
          Object {
            "push": [MockFunction],
          }
        }
        match={
          Object {
            "params": Object {
              "recipeId": 2,
            },
          }
        }
        postReview={[MockFunction]}
        setFavorite={[MockFunction]}
        singleRecipe={
          Object {
            "isFetching": false,
            "recipe": Object {
              "id": 2,
            },
            "reviews": Array [],
          }
        }
        upvoteRecipe={[MockFunction]}
        user={
          Object {
            "isAuthenticated": true,
          }
        }
      />
    </header>
    <main>
      <div>
        <Main
          handleAddReview={[Function]}
          handleChange={[Function]}
          handleDeleteReview={[Function]}
          handleDownvote={[Function]}
          handleFavorite={[Function]}
          handleUpvote={[Function]}
          handleViewMoreReviews={[Function]}
          reviewContent=""
          singleRecipe={
            Object {
              "isFetching": false,
              "recipe": Object {
                "id": 2,
              },
              "reviews": Array [],
            }
          }
        />
      </div>
      <ToastContainer
        autoClose={5000}
        bodyClassName=""
        className={null}
        closeButton={
          <DefaultCloseButton
            ariaLabel="close"
          />
        }
        closeOnClick={true}
        hideProgressBar={false}
        newestOnTop={false}
        pauseOnHover={true}
        position="top-right"
        progressClassName=""
        style={null}
        toastClassName=""
        transition={[Function]}
      />
    </main>
    <footer>
      <Footer />
    </footer>
  </div>
</div>
`;

exports[`Recipe component should render component 1`] = `
<div
  className="row"
>
  <div
    className="page-body"
  >
    <header>
      <Header
        clearReviews={[MockFunction]}
        deleteReview={[MockFunction]}
        downvoteRecipe={[MockFunction]}
        fetchRecipe={
          [MockFunction] {
            "calls": Array [
              Array [
                2,
              ],
            ],
          }
        }
        fetchReviews={
          [MockFunction] {
            "calls": Array [
              Array [
                2,
              ],
            ],
          }
        }
        handleSearchCategory={[Function]}
        history={
          Object {
            "push": [MockFunction],
          }
        }
        match={
          Object {
            "params": Object {
              "recipeId": 2,
            },
          }
        }
        postReview={[MockFunction]}
        setFavorite={[MockFunction]}
        singleRecipe={
          Object {
            "isFetching": false,
            "recipe": Object {},
            "reviews": Array [],
          }
        }
        upvoteRecipe={[MockFunction]}
        user={
          Object {
            "isAuthenticated": false,
          }
        }
      />
    </header>
    <main>
      <div>
        <div
          className="row center-align"
        >
          <div
            className="not-found"
          >
            <p
              className="page-data"
            >
              Recipe not found
            </p>
            <i
              className="material-icons large"
            >
              error
            </i>
          </div>
        </div>
      </div>
      <ToastContainer
        autoClose={5000}
        bodyClassName=""
        className={null}
        closeButton={
          <DefaultCloseButton
            ariaLabel="close"
          />
        }
        closeOnClick={true}
        hideProgressBar={false}
        newestOnTop={false}
        pauseOnHover={true}
        position="top-right"
        progressClassName=""
        style={null}
        toastClassName=""
        transition={[Function]}
      />
    </main>
    <footer>
      <Footer />
    </footer>
  </div>
</div>
`;

exports[`Recipe component should render component 2`] = `
<div
  className="row"
>
  <div
    className="page-body"
  >
    <header>
      <Header
        clearReviews={[MockFunction]}
        deleteReview={[MockFunction]}
        downvoteRecipe={[MockFunction]}
        fetchRecipe={
          [MockFunction] {
            "calls": Array [
              Array [
                2,
              ],
            ],
          }
        }
        fetchReviews={
          [MockFunction] {
            "calls": Array [
              Array [
                2,
              ],
            ],
          }
        }
        handleSearchCategory={[Function]}
        history={
          Object {
            "push": [MockFunction],
          }
        }
        match={
          Object {
            "params": Object {
              "recipeId": 2,
            },
          }
        }
        postReview={[MockFunction]}
        setFavorite={[MockFunction]}
        singleRecipe={
          Object {
            "isFetching": true,
          }
        }
        upvoteRecipe={[MockFunction]}
        user={
          Object {
            "isAuthenticated": false,
          }
        }
      />
    </header>
    <main>
      <div
        className="center-spinner center-align"
      >
        <Spinner
          size="big"
        />
      </div>
      <ToastContainer
        autoClose={5000}
        bodyClassName=""
        className={null}
        closeButton={
          <DefaultCloseButton
            ariaLabel="close"
          />
        }
        closeOnClick={true}
        hideProgressBar={false}
        newestOnTop={false}
        pauseOnHover={true}
        position="top-right"
        progressClassName=""
        style={null}
        toastClassName=""
        transition={[Function]}
      />
    </main>
    <footer>
      <Footer />
    </footer>
  </div>
</div>
`;

exports[`Recipe component should render component 3`] = `
<div
  className="row"
>
  <div
    className="page-body"
  >
    <header>
      <Header
        clearReviews={[MockFunction]}
        deleteReview={[MockFunction]}
        downvoteRecipe={[MockFunction]}
        fetchRecipe={
          [MockFunction] {
            "calls": Array [
              Array [
                2,
              ],
            ],
          }
        }
        fetchReviews={
          [MockFunction] {
            "calls": Array [
              Array [
                2,
              ],
            ],
          }
        }
        handleSearchCategory={[Function]}
        history={
          Object {
            "push": [MockFunction],
          }
        }
        match={
          Object {
            "params": Object {
              "recipeId": 2,
            },
          }
        }
        postReview={[MockFunction]}
        setFavorite={[MockFunction]}
        singleRecipe={
          Object {
            "hasMoreReviews": true,
            "isFetching": false,
            "recipe": Object {
              "category": "Main",
              "description": "Tasty beans",
              "directions": "Do this. Do that",
              "downvotes": 0,
              "favorites": 8,
              "id": 2,
              "ingredients": "Beans, onions, palm oil, salt",
              "preparationTime": 85,
              "title": "Beans",
              "upvotes": 4,
              "views": 25,
            },
            "reviews": Array [
              Object {
                "content": "Nice recipe",
              },
            ],
          }
        }
        upvoteRecipe={[MockFunction]}
        user={
          Object {
            "isAuthenticated": false,
          }
        }
      />
    </header>
    <main>
      <div>
        <Main
          handleAddReview={[Function]}
          handleChange={[Function]}
          handleDeleteReview={[Function]}
          handleDownvote={[Function]}
          handleFavorite={[Function]}
          handleUpvote={[Function]}
          handleViewMoreReviews={[Function]}
          reviewContent=""
          singleRecipe={
            Object {
              "hasMoreReviews": true,
              "isFetching": false,
              "recipe": Object {
                "category": "Main",
                "description": "Tasty beans",
                "directions": "Do this. Do that",
                "downvotes": 0,
                "favorites": 8,
                "id": 2,
                "ingredients": "Beans, onions, palm oil, salt",
                "preparationTime": 85,
                "title": "Beans",
                "upvotes": 4,
                "views": 25,
              },
              "reviews": Array [
                Object {
                  "content": "Nice recipe",
                },
              ],
            }
          }
        />
      </div>
      <ToastContainer
        autoClose={5000}
        bodyClassName=""
        className={null}
        closeButton={
          <DefaultCloseButton
            ariaLabel="close"
          />
        }
        closeOnClick={true}
        hideProgressBar={false}
        newestOnTop={false}
        pauseOnHover={true}
        position="top-right"
        progressClassName=""
        style={null}
        toastClassName=""
        transition={[Function]}
      />
    </main>
    <footer>
      <Footer />
    </footer>
  </div>
</div>
`;

exports[`Recipe component should simulate instance methods call 1`] = `
<div
  className="row"
>
  <div
    className="page-body"
  >
    <header>
      <Header
        clearReviews={[MockFunction]}
        deleteReview={
          [MockFunction] {
            "calls": Array [
              Array [
                2,
                1,
              ],
            ],
          }
        }
        downvoteRecipe={
          [MockFunction] {
            "calls": Array [
              Array [
                2,
              ],
            ],
          }
        }
        fetchRecipe={
          [MockFunction] {
            "calls": Array [
              Array [
                2,
              ],
            ],
          }
        }
        fetchReviews={
          [MockFunction] {
            "calls": Array [
              Array [
                2,
              ],
              Array [
                2,
                5,
                5,
              ],
            ],
          }
        }
        handleSearchCategory={[Function]}
        history={
          Object {
            "push": [MockFunction] {
              "calls": Array [
                Array [
                  Object {
                    "pathname": "/catalog",
                    "search": "?search=category&list=undefined",
                  },
                ],
              ],
            },
          }
        }
        match={
          Object {
            "params": Object {
              "recipeId": 2,
            },
          }
        }
        postReview={
          [MockFunction] {
            "calls": Array [
              Array [
                2,
                Object {
                  "content": "Change in input field",
                },
              ],
            ],
          }
        }
        setFavorite={
          [MockFunction] {
            "calls": Array [
              Array [
                2,
              ],
            ],
          }
        }
        singleRecipe={
          Object {
            "isFetching": false,
            "recipe": Object {
              "id": 2,
            },
            "reviews": Array [],
          }
        }
        upvoteRecipe={
          [MockFunction] {
            "calls": Array [
              Array [
                2,
              ],
            ],
          }
        }
        user={
          Object {
            "isAuthenticated": true,
          }
        }
      />
    </header>
    <main>
      <div>
        <Main
          handleAddReview={[Function]}
          handleChange={[Function]}
          handleDeleteReview={[Function]}
          handleDownvote={[Function]}
          handleFavorite={[Function]}
          handleUpvote={[Function]}
          handleViewMoreReviews={[Function]}
          reviewContent=""
          singleRecipe={
            Object {
              "isFetching": false,
              "recipe": Object {
                "id": 2,
              },
              "reviews": Array [],
            }
          }
        />
      </div>
      <ToastContainer
        autoClose={5000}
        bodyClassName=""
        className={null}
        closeButton={
          <DefaultCloseButton
            ariaLabel="close"
          />
        }
        closeOnClick={true}
        hideProgressBar={false}
        newestOnTop={false}
        pauseOnHover={true}
        position="top-right"
        progressClassName=""
        style={null}
        toastClassName=""
        transition={[Function]}
      />
    </main>
    <footer>
      <Footer />
    </footer>
  </div>
</div>
`;
